recipes/glib/all/conanfile.py

```diff
@@ -88,91 +92,108 @@ def requirements(self):
             # for Linux, gettext is provided by libc
             self.requires("libgettext/0.21")
 
-        if tools.is_apple_os(self.settings.os):
+        if is_apple_os(self.settings.os):
             self.requires("libiconv/1.17")
 
     def validate(self):
         if hasattr(self, 'settings_build') and cross_building(self, skip_x64_x86=True):
             raise ConanInvalidConfiguration("Cross-building not implemented")
-        if tools.Version(self.version) >= "2.69.0" and not self.options.with_pcre:
+        if Version(self.version) >= "2.69.0" and not self.options.with_pcre:
             raise ConanInvalidConfiguration("option glib:with_pcre must be True for glib >= 2.69.0")
-        if self.settings.os == "Windows" and not self.options.shared and tools.Version(self.version) < "2.71.1":
+        if self.settings.os == "Windows" and not self.options.shared and Version(self.version) < "2.71.1":
             raise ConanInvalidConfiguration(
                 "glib < 2.71.1 can not be built as static library on Windows. "
                 "see https://gitlab.gnome.org/GNOME/glib/-/issues/692"
             )
-        if tools.Version(self.version) < "2.67.0" and not is_msvc(self) and not self.options.with_elf:
+        if Version(self.version) < "2.67.0" and not is_msvc(self) and not self.options.with_elf:
             raise ConanInvalidConfiguration("libelf dependency can't be disabled in glib < 2.67.0")
 
     def build_requirements(self):
         self.build_requires("meson/0.61.2")
         self.build_requires("pkgconf/1.7.4")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
-
-    @functools.lru_cache(1)
-    def _configure_meson(self):
-        meson = Meson(self)
-        defs = {}
-        if tools.is_apple_os(self.settings.os):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def layout(self):
+        self.folders.build = self._build_subfolder
+        self.folders.source = self._source_subfolder
+
+    def generate(self):
+        if is_apple_os(self.settings.os):
+            self.conf["tools.apple:sdk_path"] = apple_sdk_path(self)
+
+        tc = PkgConfigDeps(self)
+        tc.generate()
+        tc = AutotoolsDeps(self)
+        # bug? meson toolchain doesn't read CPPFLAGS
+        cppflags = tc.vars().get("CPPFLAGS")
+        tc.environment.append('CFLAGS', cppflags)
+        tc.environment.append('CXXFLAGS', cppflags)
+        # conan or meson bug? LIBPATH is ignored
```

Comment:
Is it worth opening an issue in the main repo to track?

Reply:
I think that all those problems were resolved by https://github.com/conan-io/conan/pull/11557 (regarding the issue itself https://github.com/conan-io/conan/issues/11398).
Perhaps, it's worth starting to try out the new `MesonDeps` (Conan 1.51) generator for this case. See more information at https://docs.conan.io/en/latest/reference/conanfile/tools/meson/mesondeps.html.

Reply:
https://github.com/conan-io/conan/issues/11398